Cleanup MachineDeployments when deleting a cluster

alvaroaleman opened the pull request:
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:
<!-- Add links to the related documentation changes related to this pull request. E.g. the link to the kubermatic/docs pullrequest. -->

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If  no release note is required, just write "NONE".
-->
```release-note
NONE
```

alvaroaleman committed:
Cleanup MachineDeployments when deleting a cluster

mrIncompetent commented on api/pkg/controller/cluster/deleting.go:
```diff
@@ -79,6 +79,16 @@ func (cc *Controller) deletingNodeCleanup(c *kubermaticv1.Cluster) (*kubermaticv
+	}
+	if len(machineDeploymentList.Items) > 0 {
+		if err := machineClient.Cluster().MachineDeployments(metav1.NamespaceSystem).DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{}); err != nil {
+			return nil, fmt.Errorf("failed to delete cluster MachineDeployments: %v", err)
```
Error message => Could not delete MachineDeployments in user cluster

mrIncompetent replied:
Or: failed to....

To stick to the existing pattern

alvaroaleman replied:
That is exactly what I did, if you look at the lines below, the error message there is `failed to list cluster machines: %v`

mrIncompetent commented on api/pkg/controller/cluster/deleting.go:
```diff
@@ -79,6 +79,16 @@ func (cc *Controller) deletingNodeCleanup(c *kubermaticv1.Cluster) (*kubermaticv
 
+	machineDeploymentList, err := machineClient.ClusterV1alpha1().MachineDeployments(metav1.NamespaceSystem).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list cluster MachineDeployments: %v", err)
```
Error message => Could not list MachineDeployments from user cluster

mrIncompetent replied:
Or: failed to....

To stick to the existing pattern

alvaroaleman replied:
That is exactly what I did, if you look at the lines below, the error message there is `failed to ldelete cluster machines: %v`

mrIncompetent replied:
I find the "cluster *" a bit confusing. We use "user cluser" everywhere else to express which cluster is affected

alvaroaleman replied:
Well I am fine with changing it but then we should change it anywhere. So I will just replace the `cluster` everywhere with `user cluster`?

alvaroaleman committed:
Delete MachineSets as well

alvaroaleman committed:
Wait after deleting MachineDeployments and Sets until they are gone

alvaroaleman commented:
@mrIncompetent PTAL, I have updated the error messages and changed the code, to wait for for MachineDeployments and MachineSets to be gone after deleting them

alvaroaleman committed:
Do not block controller workers when waiting for machinedeployment and machineset to be gone

alvaroaleman committed:
Remove unused consts
